src/node_data.cpp

```diff
@@ -256,9 +256,21 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+
+  bool found = false;
```

Comment:
On the plus side, get(key) is much faster with vector!
and that's going to dominate access pattern